### PR TITLE
Feature 105 side wide search

### LINKED ISSUE #105: Side wide search

### PR DESCRIPTION
**Pull request by platonovsimeon**
Resolves #105

### CODE REVIEW

File: src/components/BasicLayout.tsx [hunk 1]
```diff
@@ -137,10 +168,109 @@ class BasicLayout extends React.Component<Props, State> {
...
+
+    const pageOptions = pages.results.map((result: any) => ({ title: result.title.raw, type: strings.pages, url: result.url.raw }));
+    const newsOptions = news.results.map((result: any) => ({ title: result.title.raw, type: strings.news, url: result.url.raw }));
+    const oppiminenOptions = oppiminenPages.results.map((result: any) => ({ title: result.title.raw, type: strings.oppiminen, url: result.url.raw }));
+
+    const options = pageOptions.concat(newsOptions).concat(oppiminenOptions);
+    this.setState({ options });
+  }
+
+  private renderSearchbar = () => {
```
Comment (anttileppa) [inline src/components/BasicLayout.tsx:233]: Add tsdocs

File: src/components/pages/SearchResultsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+import React from "react";
+import strings from "../../localization/strings";
+import BasicLayout from "../BasicLayout";
+import { History } from "history";
+import ArrowLeft from "@material-ui/icons/ArrowLeft";
+import ArrowRight from "@material-ui/icons/ArrowRight";
+import CircularProgress from '@material-ui/core/CircularProgress';
+import * as qs from "query-string";
+
+interface Props {
```
Comment (anttileppa) [inline src/components/pages/SearchResultsPage.tsx:11]: Add tsdocs

File: src/components/pages/SearchResultsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+  lang: string;
+  history: History;
+  query: string;
+}
+
+interface State {
```
Comment (anttileppa) [inline src/components/pages/SearchResultsPage.tsx:17]: Add tsdocs

File: src/components/pages/SearchResultsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+  results: SearchResult[];
+  query: string;
+  searchAgain: boolean;
+  currentPage: number;
+  numberOfPages: number;
+  loading: boolean;
+  selectedResultType: SearchResultType;
+};
+
+interface SearchResult {
```
Comment (anttileppa) [inline src/components/pages/SearchResultsPage.tsx:27]: Add tsdocs

File: webpack.config.js [hunk 5]
```diff
@@ -2,7 +2,7 @@ const path = require('path');
 const webpack = require('webpack');
 
 module.exports = {
-  mode: "production",
```
Comment (anttileppa) [inline webpack.config.js:5]: revert this change

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: package.json [hunk 6]
```diff
@@ -6,11 +6,11 @@
     "@babel/polyfill": "^7.6.0",
     "@material-ui/core": "^4.5.1",
     "@material-ui/icons": "^4.5.1",
+    "@material-ui/lab": "^4.0.0-alpha.61",
     "@types/react-html-parser": "^2.0.1",
     "@types/react-swipeable-views": "^0.13.0",
     "classnames": "^2.2.6",
     "localized-strings": "^0.2.4",
-    "material-ui": "^0.20.2",
```
[the remark this answers is not in the document]
Comment (platonovsimeon) [reply package.json:13]: Yes

File: src/components/BasicLayout.tsx [hunk 7]
```diff
@@ -30,6 +35,15 @@ interface State {
   postThumbnail: string;
   eventCalendarUrl?: string;
   showMenu: boolean;
+  options: SearchOption[];
+  search: SearchOption;
+  redirectToSearch: boolean;
+}
+
+interface SearchOption {
```
[the remark this answers is not in the document]
Comment (platonovsimeon) [reply src/components/BasicLayout.tsx:46]: Fixed

File: src/components/BasicLayout.tsx [hunk 8]
```diff
@@ -137,10 +168,109 @@ class BasicLayout extends React.Component<Props, State> {
...
+        ]
+      }
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+      'Authorization': `Bearer ${elasticKey}`
+    }
+  });
+
+  private onSearchChange = async (_: React.ChangeEvent<{}>, value: string) => {
```
[the remark this answers is not in the document]
Comment (platonovsimeon) [reply src/components/BasicLayout.tsx:215]: Fixed

File: src/components/BasicLayout.tsx (hunk 1 above)
Comment (platonovsimeon) [reply src/components/BasicLayout.tsx:245]: Fixed

File: src/components/pages/SearchResultsPage.tsx (hunk 2 above)
Comment (platonovsimeon) [reply src/components/pages/SearchResultsPage.tsx:13]: Fixed

File: src/components/pages/SearchResultsPage.tsx (hunk 3 above)
Comment (platonovsimeon) [reply src/components/pages/SearchResultsPage.tsx:22]: Fixed

File: src/components/pages/SearchResultsPage.tsx (hunk 4 above)
Comment (platonovsimeon) [reply src/components/pages/SearchResultsPage.tsx:35]: Fixed

File: webpack.config.js (hunk 5 above)
Comment (platonovsimeon) [reply webpack.config.js]: Fixed

File: src/components/BasicLayout.tsx [hunk 9]
```diff
@@ -137,10 +171,118 @@ class BasicLayout extends React.Component<Props, State> {
...
+        groupBy={ option => option.type }
+        onInputChange={ this.onSearchChange } 
+        renderGroup={ this.renderGroup }
+        renderInput={ params => <TextField {...params} label={ strings.search } variant="outlined"/> }
+        renderOption={ this.renderOption }
+      />
+    );
+  }
+
+  private renderGroup = (params: any) => {
```
Comment (anttileppa) [inline src/components/BasicLayout.tsx:263]: Add tsdocs

File: src/components/BasicLayout.tsx [hunk 10]
```diff
@@ -137,10 +171,118 @@ class BasicLayout extends React.Component<Props, State> {
...
+      <>
+      <Typography style={{ marginLeft: "10px", fontSize: "1.25rem", fontWeight: "bold" }}>{ params.group }</Typography>
+      <Container style={{ overflow: "hidden", paddingLeft: 0 }}>
+        { params.children }
+      </Container>
+      </>
+    );
+  }
+
+  private renderOption = (option: SearchOption) => {
```
Comment (anttileppa) [inline src/components/BasicLayout.tsx:274]: Add tsdocs

**Review by anttileppa — APPROVED** [review]

**Pull request merged**